test/perf/perf_simple_query.cc

```diff
@@ -171,7 +199,238 @@ static schema_ptr make_counter_schema(std::string_view ks_name) {
+                    },
+                    ":C2": {
+                        "S": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+                    },
+                    ":C3": {
+                        "S": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+                    },
+                    ":C4": {
+                        "S": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+                    }
+                },
+                "ReturnValues": "NONE"
+            }
+        )";
+        auto key = std::to_string(make_random_seq(cfg));
+        // Chunked content is used to minimize string copying, and thus extra allocations
+        rjson::chunked_content content;
+        content.reserve(3);
+        content.emplace_back(prefix.data(), prefix.size(), deleter{});
+        content.emplace_back(key.data(), key.size(), deleter{});
+        content.emplace_back(postfix.data(), postfix.size(), deleter{});
+        return executor.update_item(state, tracing::trace_state_ptr(), empty_service_permit(), rjson::parse(std::move(content))).discard_result();
+    }, cfg.concurrency, cfg.duration_in_seconds, cfg.operations_per_shard);
+}
+
+static std::vector<perf_result> test_alternator_delete(service::client_state& state, noncopyable_function<void()> flush_memtables,
+        alternator::executor& executor, test_config& cfg) {
+    create_alternator_partitions(state, std::move(flush_memtables), executor, cfg);
+    return time_parallel([&] {
+        std::string json = R"(
+            {
+                "TableName": "alternator_table",
+                "Key": {
+                    "p": {
+                        "S": ")" + std::to_string(make_random_seq(cfg)) + R"("
+                    }
+                },
+                "ReturnValues": "NONE"
+            }
+        )";
+        return executor.delete_item(state, tracing::trace_state_ptr(), empty_service_permit(), rjson::parse(json)).discard_result();
+    }, cfg.concurrency, cfg.duration_in_seconds, cfg.operations_per_shard);
+}
+
+static std::vector<perf_result> do_alternator_test(std::string isolation_level,
+        service::client_state& state,
+        sharded<cql3::query_processor>& qp,
+        sharded<service::migration_manager>& mm,
+        test_config& cfg) {
+    assert(cfg.frontend == test_config::frontend_type::alternator);
+
+    sharded<alternator::executor> executor;
+    sharded<db::system_distributed_keyspace> dummy_sdks;
+    sharded<cdc::metadata> dummy_cdc_metadata;
+
+    smp_service_group_config c;
+    c.max_nonlocal_requests = 5000;
+    smp_service_group ssg = create_smp_service_group(c).get0();
+    auto& proxy = qp.local().proxy().container();
+    auto flush_memtables = [&proxy] {
+        std::cout << "Flushing partitions..." << std::endl;
+        proxy.local().get_db().invoke_on_all(&database::flush_all_memtables).get();
+    };
+    dummy_sdks.start(std::ref(qp), std::ref(mm), std::ref(proxy)).get();
+    dummy_cdc_metadata.start().get();
+    executor.start(std::ref(proxy),
```

Comment:
All this really does look like alternator_test_env. It will be nice to reduce the dependencies here.